assets/artifact-hub-pkg/package.go

```diff
@@ -190,7 +192,22 @@ func writePatternFile(pattern CatalogPattern, patternType, patternInfo, patternC
 	format := "2006-01-02 15:04:05Z"
 	currentDateTime, err := time.Parse(format, time.Now().UTC().Format(format))
 
-	artifactHubPkg := catalog.BuildArtifactHubPkg(pattern.Name, filepath.Join(dir, "design.yml"), pattern.UserID, pattern.Version, currentDateTime.Format(time.RFC3339), &pattern.CatalogData)
+	version := pattern.CatalogData.PublishedVersion
+	if version == "" {
+		version = semver.New(0, 0, 1, "", "").String()
+	} else {
+		v, err := semver.NewVersion(version)
+		if err != nil {
+			return meshkitErrors.New(ErrInvalidVersionCode, meshkitErrors.Alert,
+				[]string{"Failed to parse the version"},
+				[]string{fmt.Sprintf("The provided version '%s' is invalid.\nError: %v", version, err)},
+				[]string{"The version string is not in a valid semantic version format"},
+				[]string{"Ensure the version string follows semantic versioning format", "Check for typos or incorrect format"})
+		}
```

Comment:
No need to increment the version.
As `published_version` is the version user has selected to publish.
Increment will lead to ambiguity/discrepancy.

Also, we are sure that the `published_version` will be a valid semver version, otherwise the designs do not get persist. So we can remove the check here.

Reply:
@MUzairS15, in order for this to be true our existing data must be cleaned up. Right, @Yashsharma1911?

Reply:
I see, I understand. Should I delete the code and test it after existing data is cleaned up?